processor/src/main/java/org/mapstruct/ap/internal/model/assignment/AdderWrapper.java

```diff
@@ -61,7 +61,7 @@ public AdderWrapper( Assignment decoratedAssignment, List<Type> thrownTypesToExc
     public Set<Type> getImportTypes() {
         Set<Type> imported = new HashSet<Type>();
         imported.addAll( super.getImportTypes() );
-        imported.add( getSourceType() );
+        imported.add( getSourceType().getTypeParameters().get( 0 ) );
```

Comment:
Thanks for the explanation. As I wrote in the other comment, I had problem figuring out how it was created before. You explained it well with:

>In the old solution, the SourceType in SourceRHS was simply replaced by the type of the collection element when an ADDER was applicable.